src/Workspaces/Core/Portable/SymbolKey/SymbolKeyResolution.cs

```diff
@@ -1,36 +1,98 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
+using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
-    internal struct SymbolKeyResolution
+    internal struct SymbolKeyResolution : IEquatable<SymbolKeyResolution>
```

Comment:
i believe this type is just SymbolInfo.  However, i couldn't use that because it didn't have public entrypoints.  If we're now going and making this API public, we should just use SymbolInfo and get rid of this.

Reply:
Yeah, this is just `SymbolInfo`. However, `SymbolInfo` still doesn't have public entry points. :smile:

I haven't yet convinced myself that there isn't something else that might need to go on this type yet.

Reply:
PArt of hte reason @gafter would be good to talk to is that he just spent a lot of time thinking about naming equivalence of named-types across compilations.  I *believe* that his form of eqiuvalence matches the form we have in symbol key.  But it would be good to make sure and to do things in a consistent way.

Effectively, we're codifying what names means, and i think that's a *great* thing.  But we shoudl make sure we're all being consistent here.  I see no reason we can't be.  But we should be careful and conscientious here :)

Reply:
Note that the `IsSymbolAccessibleWithin` API that is being added to the compiler (https://github.com/dotnet/roslyn/pull/27312) does not work cross-compilation or deal with symbol equivalence. During the design process, we determined that the cross-compilation scenarios were not the most important for adding the API. Additionally, we decided that asking "cross compilation symbol equivalence" questions are really more speculative in nature, so they should appear as speculative APIs if they're going to be at compiler layer.

Reply:
> does not work cross-compilation or deal with symbol equivalence

Really?  Then how could it replace the IDE version we have?  Cross-compilation was a critical part of it because we def did things like check if a VB symbol was accessible to some c# compilation... :)

> Additionally, we decided that asking "cross compilation symbol equivalence" questions are really more speculative in nature, so they should appear as speculative APIs if they're going to be at compiler layer.

that's totally reasonable to me.  I still think this should move down :)   But it should def have the right shape if it is at the compiler layer.  Sounds like a lot of good discussions and design ideas have been made about that sort of thing :)

Reply:
> Really? Then how could it replace the IDE version we have? Cross-compilation was a critical part of it because we def did things like check if a VB symbol was accessible to some c# compilation... :)

I did a survey of the source and found that most of the uses of the existing API actually did not work cross-compilation. For example, where it's used in completion and signature help. Feature that still need to use it in a cross-compilation fashion will still need to use the old API. Or, we'll need to add a speculative version of the API for those features to use.

Reply:
> I did a survey of the source and found that most of the uses of the existing API actually did not work cross-compilation.

Hrmm.. i thought we used it in several code-gen features (like gen constructor) to figure out things.  And don't we use it for add-using to see if we can add a p2p reference?  It's also used in a ton of extensions that themselves build up into other features, right?

I think, if it's not intended to be used cross-compilation, we should at least try to assert in our impl and see if that causes anything to break.  :)